test/jasmine/tests/transform_groupby_test.js

```diff
@@ -0,0 +1,523 @@
+
+                    expect(gd._fullData[1].ids).toEqual(['r', 'y', 'u']);
+                    expect(gd._fullData[1].x).toEqual([-2, 1, 2]);
+                    expect(gd._fullData[1].y).toEqual([2, 5, 4]);
+                    expect(gd._fullData[1].marker.line.width).toEqual([4, 2, 3]);
+
+                    assertDims([4, 3]);
+
+                    done();
+                });
+            };
+        }
+
+        // basic test
+        var mockData1 = [{
+            mode: 'markers',
+            ids: ['q', 'w', 'r', 't', 'y', 'u', 'i'],
+            x: [1, -1, -2, 0, 1, 2, 3],
+            y: [0, 1, 2, 3, 5, 4, 6],
+            marker: {line: {width: [4, 2, 4, 2, 2, 3, 3]}},
+            transforms: [{
+                type: 'groupby',
+                groups: ['a', 'a', 'b', 'a', 'b', 'b', 'a'],
+                style: { a: {marker: {color: 'red'}}, b: {marker: {color: 'blue'}} }
+            }]
+        }];
+
+        // heterogenously present attributes
+        var mockData2 = [{
+            mode: 'markers',
+            ids: ['q', 'w', 'r', 't', 'y', 'u', 'i'],
+            x: [1, -1, -2, 0, 1, 2, 3],
+            y: [0, 1, 2, 3, 5, 4, 6],
+            marker: {line: {width: [4, 2, 4, 2, 2, 3, 3]}},
+            transforms: [{
+                type: 'groupby',
+                groups: ['a', 'a', 'b', 'a', 'b', 'b', 'a'],
+                style: {
+                    a: {
+                        marker: {
+                            color: 'orange',
+                            size: 20,
+                            line: {
+                                color: 'red'
+                            }
+                        }
+                    },
+                    b: {
+                        mode: 'markers+lines', // heterogeonos attributes are OK: group 'a' doesn't need to define this
+                        marker: {
+                            color: 'cyan',
+                            size: 15,
+                            line: {
+                                color: 'purple'
+                            },
+                            opacity: 0.5,
+                            symbol: 'triangle-up'
+                        },
+                        line: {
+                            color: 'purple'
+                        }
+                    }
+                }
+            }]
+        }];
+
+        // attributes set at top level and partially overridden in the group item level
+        var mockData3 = [{
+            mode: 'markers+lines',
+            ids: ['q', 'w', 'r', 't', 'y', 'u', 'i'],
+            x: [1, -1, -2, 0, 1, 2, 3],
+            y: [0, 1, 2, 3, 5, 4, 6],
+            marker: {
+                color: 'darkred', // general 'default' color
+                line: {
+                    width: [4, 2, 4, 2, 2, 3, 3],
+                    // a general, not overridden array will be interpreted per group
+                    color: ['orange', 'red', 'green', 'cyan']
```

Comment:
Trace belonging to group `a` gets the `marker.line.color` from the split array
Trace belonging to group `b` gets the `marker.line.color` from `transforms.style.b.marker.line.color`

I updated the test case to demonstrate it (will push soon).